src/utils.ts

```diff
@@ -0,0 +1,697 @@
+// Dependencies
+import chalk from 'chalk';
+import fs from 'fs';
+import path from 'path';
+import Fuse from 'fuse.js';
+import ErrorWithoutStack from './error-without-stack';
+
+// Interface declarations
+interface InputObject {
+	command: string;
+	data?: string | number;
+	[index: string]: boolean | string | number | undefined;
+}
+
+// Interface declarations
+export interface AppSettings {
+	name: string | null;
+	packageName: string | null;
+	version: string | null;
+}
+
+export interface CommandSpec {
+	data: {
+		accepts?: string[];
+		description: string;
```

Comment:
You looked into this and said that the `description` is always required; however, I looked at the code on the `master` branch and found that it's optional. It will only be displayed if present.

```suggestion
		description?: string;
```

Reply:
Hey Wes,

Look at the `master` `src/utils.js` around line 362 - my take is that the idea here is that if the supplied arguments, after having been organized, results in there being 'data', and, then, if either the spec's data is entirely null, or, the spec's data.description is null, then the code proceeds to generate an error message.  So, I think this means (unless it's a bug): if you have a spec.data, it *must* have a description (unless the object of the app is to endlessly produce usage error messages).

Additionally, if you look at `master`'s `src/screen.js` around line 100, it requires the presence of both the mergedSpec.data and the mergedSpec.data.description before it decides that the app 'acceptsData', again reinforcing the notion that within the `CommandSpec`, that if you have a `data` object, it requires a `description` property for it to be considered valid.

Of course, this is what it's doing now.  It could be changed :)

Let me know what you think.

Reply:
Interesting, there does appear to be an inconsistency. On `screens.js` [line 238](https://github.com/sparksuite/waterfall-cli/blob/master/src/screens.js#L238), it treats the description as optional, which is what I was looking at.

But the other two places you mentioned do treat it as required. We'll, therefore, need to decide what's better going forward.

I'm inclined to believe that it's better as optional; this would match the descriptions for flags, options, and commands. Plus it gives the engineer using Waterfall CLI control over whether or not they think it's necessary.

We should create a separate PR to change this. I can probably do that pretty quickly, but I'll wait to merge it to see what you think.